invenio_accounts/views/__init__.py

```diff
@@ -2,12 +2,33 @@
 #
 # This file is part of Invenio.
 # Copyright (C) 2017-2018 CERN.
+# Copyright (C)      2021 TU Wien.
 #
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Invenio-accounts views."""
 
+from flask import abort, current_app, request
+from flask_security.views import anonymous_user_required
+from flask_security.views import login as base_login
+
 from .settings import blueprint
 
-__all__ = ('blueprint', )
+
+@anonymous_user_required
+@blueprint.route("/login")
+def login(*args, **kwargs):
+    """Disable login credential submission if local login is disabled."""
+    local_login_enabled = current_app.config.get(
+        "ACCOUNTS_LOCAL_LOGIN_ENABLED", True
+    )
+
+    if request.method != "GET" and not local_login_enabled:
```

Comment:
I am not sure I understand why the `!= 'GET'` check: don't we want to return 404 also if the user goes to the web page? (maybe that's another view...)

Reply:
no, it's the same view function that handles the login page rendering on GET and the login process on POST
if we give a 404 on GET as well, that'd override the rendering of the login page
however, we still want to show the login page when there's multiple external auth services configured, but local login disabled